addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,211 @@
+        self.product_processable.image_fetch_pending = True
+        try:
+            self._fetch_image_from_google(self.product_processable[:10])
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                    fields.Datetime.now() + timedelta(days=1.0)
+                )
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'warning',
+                        'title': _("Product images"),
+                        'message': _("Custom search quota exceeded. We will start again tomorrow"),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+            elif http_error.response.status_code == requests.codes.forbidden:
+                project_url = re.search(r"(?P<url>https?://[^\s]+)",
+                            http_error.response.json().get('error').get('message')).group('url')
+                raise UserError(_("Custom Search API is not enabled in your Google project. "
+                                  "Enable it by visiting %(url)s then retry. If you enabled this "
+                                  "API recently, wait a few minutes for the action to propagate "
+                                  "to Google's systems and retry.", url=project_url))
+            elif http_error.response.status_code == requests.codes.bad_request:
+                raise UserError(_("Your API Key or your Search Engine ID is incorrect."))
+        except ConnectionError:
+            self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                fields.Datetime.now() + timedelta(hours=1.0)
+            )
+        else:
+            if self.product_processable[10:]:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger()
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'success',
+                        'title': _("Product images"),
+                        'message': _("We are processing your products. Please come back in a few "
+                                    "moments to see the images."),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+
+    def _cron_fetch_image(self):
+        """ Fetch images of a list of products using their barcode
+
+        This function is called from a cron job. If the daily request limit is reach, the cron job
+        will be schedule to run again a day later.
+        """
+        product_processable = self.env['product.product'].search([
+            ('image_fetch_pending', '=', 'True'),
+        ], limit=101)
```

Comment:
Why 101 and not 100? To trigger a new cron run if there is a remaining? If so, that's not a bad idea but it's a bit obscure, don't you think?

Reply:
That's exactly why 101 and not 100. I didn't find another solution apart from trigger the cron no matter the remaining number of products to process. In that case, the cron will always launch one time for nothing and may cause performance issues.

Reply:
Let's settle this with a short comment ;)